website/blog/2025-12-10-react-native-0.83.md

```diff
@@ -0,0 +1,272 @@
+
+- `hidden`: hides the children, unmounts effects, and defers all updates until React has nothing left to work on.
+- `visible`: shows the children, mounts effects, and allows updates to be processed normally.
+
+One interesting feature of trees hidden using `<Activity mode='hidden'>` is that they preserve their state. So, when they become visible again, they, for example, keep the search status and the selection from a previous user interaction.
+
+| React 19.1.1                                      | React 19.2.1                                   |
+| ------------------------------------------------- | ---------------------------------------------- |
+| ![](/blog/assets/0.83-react-19.2-no-activity.gif) | ![](/blog/assets/0.83-react-19.2-activity.gif) |
+
+You can read more about `<Activity>` in the [React docs](https://react.dev/reference/react/Activity).
+
+### `useEffectEvent`
+
+One common pattern with useEffect is to notify the app code about some kind of "event" from an external system. The problem with this approach is that a change to any values used inside such an event will cause the surrounding Effect to re-run.
+
+To solve this, most users disable the lint rule and exclude the dependency. But that can lead to bugs since the linter can no longer help you keep the dependencies up to date if you need to update the Effect later.
+
+With `useEffectEvent`, you can split the "event" part of this logic out of the Effect that emits it.
+
+You can read more about `useEffectEvent` in the [React docs](https://react.dev/reference/react/Activity).
+
+## New DevTools features
+
+In 0.83 we're excited to deliver some long awaited features and quality of life improvements to React Native DevTools.
+
+### Network and Performance panels
+
+Network inspection and performance tracing are two powerful new capabilities in React Native DevTools, available today.
+
+![Network panel list in React Native DevTools](/docs/assets/debugging-rndt-network.jpg)
+
+**Network inspection**, now available for all React Native apps, allows you to view and understand the network requests made by your app. Logged requests provide detailed metadata such as timings and headers sent/received, as well as response previews. And — for the first time — you can use the Initiator tab to see where in your code a network request originated.
+
+<details>
+<summary><strong>💡 Network event coverage, Expo support</strong></summary>
+
+Today, we record all network calls through `fetch()`, `XMLHttpRequest`, and `<Image>` — with support for custom networking libraries, such as Expo Fetch, coming later.
+
+Because of this, apps using Expo will continue to see the "Expo Network" panel, which will log these additional requests but has slightly reduced features such as no Performance panel integration.
```

Comment:
Let's be clearer that Expo Network is a different implementation and not the same feature.